### ISSUE TITLE
not precompile-safe

### ISSUE THREAD

**Issue by stevengj** [JuliaGL/GLVisualize.jl#15]
As described in the [Julia manual](http://docs.julialang.org/en/latest/manual/modules/#module-initialization-and-precompilation), if you want to precompile this module you need to follow certain rules.  (Basically, you need to put anything that must occur at *runtime* into an `__init__()` function, and you need to be careful to avoid initializing global variables outside of `__init__` to values that may change at runtime, e.g. pointers.)  It looks like this module has several problems that need to be fixed in order to make it precompile-safe.

Until it is precompile-safe, you should put `VERSION >= v"0.4.0-dev+6521" && __precompile__(false)` at the top of the `GLVisualize.jl` file.

Current problems include:

- [ ] `visualize/plotlibs.jl` executes code that really should go into `__init__` because it seems to depend on the callers *runtime* environment.  Also, it looks [unsafe](https://github.com/JuliaGL/GLVisualize.jl/blob/e38768cddd98d5f587553525b29cba6a0512420f/src/visualize/plotlibs.jl#L5) because it looks like it stores a `PyObject` in a global, and `PyObject` contains a pointer.
- [ ] `FreeType.init()` initializes a [global pointer](https://github.com/JuliaGL/GLVisualize.jl/blob/3cee1e7cca0729dfbfaa8b2aad459d2492f3ae0a/src/visualize/text/freetype_text.jl#L78) that should really be initialized at runtime.  Change this to `__init__`.
- [ ] `texture_atlas.jl` initializes a [global fontface](https://github.com/JuliaGL/GLVisualize.jl/blob/e38768cddd98d5f587553525b29cba6a0512420f/src/texture_atlas/texture_atlas.jl#L95) that looks like it contains a pointer.

There may be other problems, too.  (It would be a lot easier to audit the code if you did not split it up into so very many files, especially since many of the files contain only one or even zero functions.)

**Comment by SimonDanisch** [comment]
It seems like you're on master. Sorry about the first comment in JuliaLang/julia#12686 without any pointers on how to reproduce it, but in the last post I pointed out that one has to be on the ntuples branch. I think I cleaned up all points from the manual in ntuples.
Master is completely unsafe for precompilation and it was quite a big rebuild to make it save (including a complete rewrite of FixedSizeArrays) which is why its still in a different branch.
Thanks for the patience here, I will go through my code to see if I overlooked anything!
